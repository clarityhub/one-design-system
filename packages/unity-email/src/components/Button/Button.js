import React from 'react';
import {
	MjmlButton,
} from 'mjml-react';
import { node } from 'prop-types';
import colors from '@clarityhub/unity-core/dist/theme/colors';

import { variants } from '../../theme/fonts';

const { letterSpacing, ...validButtonOptions } = variants.button;

const buttonBase = Object.assign(
	{},
	validButtonOptions,
	{
		fontSize: '16px',
		borderRadius: '4px', // TODO merge with borders.borderRadius
		paddingTop: '8px',
		paddingBottom: '8px',
		paddingLeft: '20px',
		paddingRight: '20px',
	}
);

const determineColors = (type) => {
	switch (type) {
	case 'primary':
		return {
			color: colors.white,
			backgroundColor: colors.primary,
		};
	case 'danger':
		return {
			color: colors.white,
			backgroundColor: colors.danger,
		};
	case 'default':
	default:
		return {
			color: colors.primary,
			backgroundColor: 'rgba(255, 255, 255, 0)',
		};
	}
};

const Button = ({ children, center, type, ...rest }) => (
	<MjmlButton {...buttonBase} {...determineColors(type)} align={center ? 'center' : 'left'} textAlign={center ? 'left': 'center'} {...rest}>
		{children}
	</MjmlButton>
);

Button.propType = {
	children: node,
};

export default Button;
