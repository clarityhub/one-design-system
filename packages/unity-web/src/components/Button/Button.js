import React from 'react';
import cx from 'classnames';
import { css } from '@emotion/core';
import styled from '@emotion/styled';
import { bool, func, oneOf, oneOfType, node, string } from 'prop-types';
import { variants } from '@clarityhub/unity-core/lib/typography';
import borders from '@clarityhub/unity-core/lib/borders';
import colors from '@clarityhub/unity-core/lib/colors';
import { lighten, darken, desaturate } from '@clarityhub/unity-core/lib/utilities/color';

import Loading from '../svgs/Loading';

const Button = styled.button(
	css`
        overflow: visible;
        display: inline-block;

        ${variants.button.string}

        box-shadow: ${colors.shadow.default};
        cursor: pointer;
        text-align: center;
        white-space: nowrap;
        vertical-align: middle;
        user-select: none;
        border: 1px solid transparent;
        padding: .5rem 1.25rem;
        font-size: 1rem;
        line-height: 1.5;
        border-radius: ${borders.borderRadius.rem}rem;
        position: relative;
        text-decoration: none;
        text-transform: uppercase;
        transition: color .15s ease-in-out,background-color .15s ease-in-out,border-color .15s ease-in-out,box-shadow .15s ease-in-out;

        &:focus,
        &:hover {
            text-decoration: none;
        }

        &:focus,
        &.focused {
            outline: 0;
            box-shadow: 0 0 0 .2rem ${colors.focused.default};
        }

        &:disabled,
        &.disabled {
            cursor: not-allowed;
            pointer-events: none;
        }

        &:not(:disabled):not(.disabled) {
            cursor: pointer;
        }

        svg {
            height: 1rem;
            width: 1rem;
        }

        --text: ${colors.primary.default};
        --text-active: ${colors.black.default};
        --bg: transparent;
        --bg-active: transparent;
        --bg-disabled: transparent;

        &.-default,
        &.-primary,
        &.-danger,
        &.-white {
            color: var(--text);
            background-color: var(--bg);
            border-color: var(--bg);

            &.-outline {
                border-color: var(--outline, var(--bg));
                color: var(--outline-text, var(--bg));
            }

            &.-text {
                color: var(--text-text, var(--bg));
            }

            &:hover,
            &.active {
                background-color: var(--bg-active);
                border-color: var(--bg-active);
                color: var(--text-active);

                &.-outline {
                    background-color: var(--bg) !important;
                    border: 1px solid var(--bg);
                    color: var(--outline-text-active, var(--text));
                }

                &.-text {
                    color: var(--text-text-active, var(--bg-active));
                }
            }

            &:disabled,
            &.disabled {
                background-color: var(--bg-disabled);
                border-color: var(--bg-disabled);
                color: var(--text);

                &.-outline {
                    border: 1px solid var(--outline-disabled, var(--outline-text-disabled, var(--bg-disabled)));
                    color: var(--outline-text-disabled, var(--bg-disabled));
                }

                &.-text {
                    color: var(--text-text-disabled, var(--bg-disabled));
                }
            }
        }

        &.-primary {
            --text: ${colors.white.default};
            --text-active: ${colors.white.default};
            --outline-disabled: ${colors.gray.default};
            --bg: ${colors.primary.default};
            --bg-active: ${desaturate(darken(colors.primary.default))};
            --bg-disabled: ${desaturate(lighten(colors.primary.default))};
        }

        &.-danger {
            --text: ${colors.white.default};
            --text-active: ${colors.white.default};
            --outline-disabled: ${colors.gray.default};
            --bg: ${colors.danger.default};
            --bg-active: ${desaturate(darken(colors.danger.default))};
            --bg-disabled: ${lighten(colors.danger.default, 0.3)};
        }

        &.-default {
            --text: ${colors.black.default};
            --text-active: ${colors.black.default};
            --outline: ${colors.black.default};
            --outline-text: ${colors.black.default};
            --outline-text-active: ${colors.black.default};
            --outline-text-disabled: ${colors.gray.default};
            --text-text: ${colors.black.default};
            --text-text-active: ${colors.black.default};
            --text-text-disabled: ${colors.gray.default};
            --bg: ${colors.white.default};
            --bg-active: ${colors.dove.default};
            --bg-disabled: ${colors.muted.default};
        }

        &.-white {
            --text: ${colors.primary.default};
            --text-active: ${colors.black.default};
            --outline: ${colors.white.default};
            --outline-text: ${colors.white.default};
            --outline-text-active: ${colors.primary.default};
            --outline-text-disabled: ${colors.gray.default};
            --text-text: ${colors.white.default};
            --text-text-active: ${colors.white.default};
            --text-text-disabled: ${colors.gray.default};
            --bg: ${colors.white.default};
            --bg-active: ${colors.dove.default};
            --bg-disabled: ${colors.muted.default};
        }

        &.-text {
            background-color: transparent !important;
            border: 1px solid transparent;
            box-shadow: none;

            &:hover,
            &.active {
                border-color: transparent;
            }

            &.disabled,
            &:disabled {
                border: 1px solid transparent;
            }
        }

        &.-outline {
            background-color: transparent !important;
            border: 1px solid inherit;
            box-shadow: none;
        }

        &.disabled,
        &:disabled {
            cursor: not-allowed;
        }
    `,
	({ size }) => {
		switch (size) {
		case 'small':
			return css`
                font-size: 0.9rem;
                padding: .2rem 1rem;
            `;
		case 'large':
			return css`
                font-size: 1.2rem;
                padding: 0.8rem 1.4rem;
            `;
		case 'default':
		default:
			return '';
		}
	},

	({ block }) => block && css`
        display: block;
        width: 100%;
    `,

	({ loading }) => loading && css`
        color: transparent !important;
        cursor: wait !important;

        &:hover {
            color: transparent !important;
        }
    `,
);

const IconWrapper = styled.span`
    font-size: inherit;
    margin-right: 0.3rem;
    vertical-align: middle;

    svg {
        height: 1rem;
        width: 1rem;

        ${({ size }) => {
		switch (size) {
		case 'small':
			return css`
                height: 0.9rem;
                width: 0.9rem;
            `;
		case 'large':
			return css`
                height: 1.2rem;
                width: 1.2rem;
            `;
		case 'default':
		default:
			return '';
		}
	}}
    }
`;

const LoadingWrapper = styled.div`
    position: absolute;
    left: 50%;
    transform: translate(-50%, -50%);
    top: 50%;
    align-content: center;
    justify-content: center;
    display: flex;

    svg {
        height: 1.4rem;
        width: 1.4rem;

        ${({ size }) => {
		switch (size) {
		case 'small':
			return css`
                height: 1.2rem;
                width: 1.2rem;
            `;
		case 'large':
			return css`
                height: 2.4rem;
                width: 2.4rem;
            `;
		case 'default':
		default:
			return '';
		}
	}}

    path {
        ${({ type, outline, text }) => {
		switch (type) {
		case 'primary':
			return css`
                fill: ${colors.white.default};

                ${outline && css`
                    fill: ${colors.primary.default};
                `}

                ${text && css`
                    fill: ${colors.primary.default};
                `}

                &:hover,
                &.active {
                    ${outline && css`
                        fill: ${colors.white.default};
                    `}

                    ${text && css`
                        fill: ${desaturate(darken(colors.primary.default))};
                    `}
                }

                &:disabled,
                &.disabled {
                    ${outline && css`
                        fill: ${desaturate(lighten(colors.primary.default))};
                    `}

                    ${text && css`
                        fill: ${desaturate(lighten(colors.primary.default))};
                    `}
                }
            `;
		case 'danger':
			return css`
                fill: ${colors.white.default};

                ${outline && css`
                    fill: ${colors.danger.default};
                `}

                ${text && css`
                    fill: ${colors.danger.default};
                `}

                &:hover,
                &.active {
                    ${outline && css`
                        fill: ${colors.white.default};
                    `}

                    ${text && css`
                        fill: ${desaturate(darken(colors.danger.default))};
                    `}
                }

                &:disabled,
                &.disabled {
                    ${outline && css`
                        fill: ${desaturate(lighten(colors.danger.default, 0.3))};
                    `}

                    ${text && css`
                        fill: ${desaturate(lighten(colors.danger.default, 0.3))};
                    `}
                }
            `;
		case 'default':
		default:
			return css`
                fill: ${colors.black.default};

                ${outline && css`
                    fill: ${colors.black.default};
                `}

                ${text && css`
                    fill: ${colors.black.default};
                `}

                &:hover,
                &.active {
                    ${outline && css`
                        fill: ${colors.black.default};
                    `}

                    ${text && css`
                        fill: ${desaturate(darken(colors.black.default))};
                    `}
                }

                &:disabled,
                &.disabled {
                    fill: ${darken(colors.gray.default, 0.1)};

                    ${outline && css`
                        fill: ${lighten(colors.gray.default, 0.01)};
                    `}

                    ${text && css`
                        fill: ${desaturate(lighten(colors.black.default))};
                    `}
                }
            `;
		}}
}
        }
    }
`;

const renderIcon = (Icon) => {
	if (typeof Icon === 'function') {
		return <Icon />;
	}

	return Icon;
};

const ButtonWrapper = ({
	className,
	type,
	outline,
	text,
	size,
	icon,
	children, 
	component,
	loading,
	buttonType = 'button',
	...rest
}) => {
	let Component = Button;

	if (component) {
		Component = Button.withComponent(component);
	}

	return (
		<Component
			className={cx(className, {
				[`-${type}`]: type,
				'-outline': outline,
				'-text': text,
			})}
			outline={outline}
			text={text}
			size={size}
			loading={loading}
			type={buttonType}
			{...rest}
		>
			{
				icon && (
					<IconWrapper size={size}>{renderIcon(icon)}</IconWrapper>
				)
			}
			{
				loading && (
					<LoadingWrapper
						type={type}
						outline={outline}
						text={text}
						size={size}
						{...rest}
					>
						<Loading />
					</LoadingWrapper>
				)
			}
			{children}
		</Component>
	);
};

ButtonWrapper.propTypes = {
	block: bool,
	buttonType: string,
	component: oneOfType([string, func]),
	icon: oneOfType([node, func]),
	loading: bool,
	outline: bool,
	size: oneOf(['small', 'default', 'large']),
	text: bool,
	type: oneOf(['default', 'white', 'primary', 'danger', 'success']),
};

ButtonWrapper.defaultProps = {
	type: 'default',
};

export { Button as EmotionButton };
export default ButtonWrapper;
