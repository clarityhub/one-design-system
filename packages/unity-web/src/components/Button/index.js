export { default, EmotionButton } from './Button';