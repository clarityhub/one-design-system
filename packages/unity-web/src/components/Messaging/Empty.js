import React, { Fragment } from 'react';
import { arrayOf, string, node, func, oneOfType } from 'prop-types';
import Icon from '@mdi/react';

import Box from '../../scaffolding/Box';
import Typography from '../Typography/Typography';
import { mdiTextBoxRemoveOutline } from './emptyIcon';

const Empty = ({
	icon,
	message = 'No data',
	details,
	actions = [],
}) => (
	<Box direction="column" style={{ textAlign: 'center' }}>
		<Box padding="medium">
			<Icon
				path={icon || mdiTextBoxRemoveOutline}
				title={'Not found'}
				size={3}
				style={{
					margin: 'auto',
				}}
			/>
		</Box>
		{message && <Typography type="h3" noMargin noPadding>
		    {message}
		</Typography>}
		{details && <Typography>
		    {details}
		</Typography>}
        
		{actions.map((action, i) => {
			return (
				<Fragment key={i}>
					{action}
				</Fragment>
			);
		})}
	</Box>
);

Empty.propTypes = {
	actions: arrayOf(oneOfType([node, func])),
	details: oneOfType([node, func]),
	icon: string,
	message: oneOfType([node, func]),
};

export default Empty;