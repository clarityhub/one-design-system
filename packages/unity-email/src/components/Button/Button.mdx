---
menu: Components
name: Button
---

import { Playground, PropsTable } from 'docz';
import Button from './Button';

# Email Button

```js
import Button from '@clarityhub/unity-email/lib/components/Button';
```

## Prop Types

<PropsTable of={Button} />
