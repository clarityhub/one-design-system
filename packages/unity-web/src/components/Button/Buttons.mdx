---
menu: Components
name: Buttons
---

import { Playground, PropsTable } from 'docz';
import DeleteIcon from '@material-ui/icons/Delete';
import CheckIcon from '@material-ui/icons/CheckCircle';
import Button from './index';

# Buttons

Buttons are used for triggering various actions.

```js
import Button from '@clarityhub/unity-web/lib/components/Button';
```

## Button Types

There are 4 main button types: default, white, primary, and danger.

* **Default buttons** should be used for actions that will not cause any changes to happen. For example, a Cancel button may close a modal, but it will not cause any changes to happen.
* **White buttons** are used when you have a non-white background color and you need to use a default button that may have variants. A white button should normally be used the same way a default button would be used.
* **Primary buttons** are for the primary action you would like a user to take. These buttons should not have any destructive consequence to clicking them.
* **Danger buttons** are for actions that would cause destructive consequences to occur; like deleting an item or unscheduling a campaign.

<Playground>
    <div style={{ backgroundColor: '#dde' }}>
        <Button>Default</Button>
        {' '}
        <Button type="white">White</Button>
        {' '}
        <Button type="primary">Primary</Button>
        {' '}
        <Button type="danger">Danger</Button>
    </div>
</Playground>

## As a Link

If you need a React Router Link, use [LinkButton](./packages-unity-web-src-components-link-linkbutton).

Otherwise, you can have buttons become anchor tags.

<Playground>
    <Button component="a" href="#">Cancel</Button>
    {' '}
    <Button component="a" href="#" type="white">Cancel</Button>
    {' '}
    <Button component="a" href="#" type="primary">Create</Button>
    {' '}
    <Button component="a" href="#" type="danger">Delete</Button>
</Playground>

## States

<Playground>
    <Button className="focused">Focused</Button>
    {' '}
    <Button className="focused" type="white">Focused</Button>
    {' '}
    <Button className="focused" type="primary">Focused</Button>
    {' '}
    <Button className="focused" type="danger">Focused</Button>
    <br/>
    <br/>
    <Button className="active">Active</Button>
    {' '}
    <Button className="active" type="white">Active</Button>
    {' '}
    <Button className="active" type="primary">Active</Button>
    {' '}
    <Button className="active" type="danger">Active</Button>
    <br/>
    <br/>
    <Button className="disabled">Disabled</Button>
    {' '}
    <Button className="disabled" type="white">Disabled</Button>
    {' '}
    <Button className="disabled" type="primary">Disabled</Button>
    {' '}
    <Button className="disabled" type="danger">Disabled</Button>
</Playground>

## Loading

The `loading` property should be used with the `disabled` property.

<Playground>
    <Button size="small" loading disabled>Text that is pretty long</Button>
    {' '}
    <Button size="small" loading disabled type="primary">Text that is pretty long</Button>
    {' '}
    <Button size="small" loading disabled type="danger">Text that is pretty long</Button>
    <br/>
    <br/>
    <Button loading disabled>Text that is pretty long</Button>
    {' '}
    <Button loading disabled type="primary">Text that is pretty long</Button>
    {' '}
    <Button loading disabled type="danger">Text that is pretty long</Button>
    <br/>
    <br/>
    <Button size="large" loading disabled>Text that is pretty long</Button>
    {' '}
    <Button size="large" loading disabled type="primary">Text that is pretty long</Button>
    {' '}
    <Button size="large" loading disabled type="danger">Text that is pretty long</Button>
    <br/>
    <br/>
    <Button size="small" outline loading disabled>Text that is pretty long</Button>
    {' '}
    <Button size="small" outline loading disabled type="primary">Text that is pretty long</Button>
    {' '}
    <Button size="small" outline loading disabled type="danger">Text that is pretty long</Button>
    <br/>
    <br/>
    <Button outline loading disabled>Text that is pretty long</Button>
    {' '}
    <Button outline loading disabled type="primary">Text that is pretty long</Button>
    {' '}
    <Button outline loading disabled type="danger">Text that is pretty long</Button>
    <br/>
    <br/>
    <Button size="large" outline loading disabled>Text that is pretty long</Button>
    {' '}
    <Button size="large" outline loading disabled type="primary">Text that is pretty long</Button>
    {' '}
    <Button size="large" outline loading disabled type="danger">Text that is pretty long</Button>
</Playground>

## Sizes

You can change the size of buttons for use in different contexts.

* Small buttons may be used in table views
* The default size of buttons should be used in most cases
* Large buttons can be used for calls to action

<Playground>
    <Button size="small">Small</Button>
    {' '}
    <Button size="small" type="primary">Small</Button>
    {' '}
    <Button size="small" type="danger">Small</Button>
    <br/>
    <br/>
    <Button size="default">Default</Button>
    {' '}
    <Button size="default" type="primary">Default</Button>
    {' '}
    <Button size="default" type="danger">Default</Button>
    <br/>
    <br/>
    <Button size="large">Large</Button>
    {' '}
    <Button size="large" type="primary">Large</Button>
    {' '}
    <Button size="large" type="danger">Large</Button>
</Playground>

## Icon Buttons

Specifying the icon prop will space out the icon appropriately when used with text.

<Playground>
    <Button icon={DeleteIcon}>With Icon</Button>
    {' '}
    <Button icon={<DeleteIcon />} type="primary">With Icon</Button>
    {' '}
    <Button icon={<DeleteIcon />} type="danger">With Icon</Button>
    {' '}
    <Button size="large" icon={<DeleteIcon />}>With Icon</Button>
</Playground>

## Other Variants

There are outline and text variants that can be used in different scenarios.

* Outline buttons should only be used for secondary actions
* Text buttons can be used to de-emphasis an action

<Playground>
    <div style={{ backgroundColor: '#dde' }}>
        <Button outline>Outline</Button>
        {' '}
        <Button outline type="white">White</Button>
        {' '}
        <Button outline type="primary">Outline</Button>
        {' '}
        <Button outline type="danger">Outline</Button>
        <br/>
        <br/>
        <Button disabled outline>Outline</Button>
        {' '}
        <Button disabled outline type="white">White</Button>
        {' '}
        <Button disabled outline type="primary">Outline</Button>
        {' '}
        <Button disabled outline type="danger">Outline</Button>
        <br/>
        <br/>
        <Button text>text</Button>
        {' '}
        <Button text type="white">text</Button>
        {' '}
        <Button text type="primary">text</Button>
        {' '}
        <Button text type="danger">text</Button>
        <br/>
        <br/>
        <Button disabled text>text</Button>
        {' '}
        <Button disabled text type="white">text</Button>
        {' '}
        <Button disabled text type="primary">text</Button>
        {' '}
        <Button disabled text type="danger">text</Button>
        <br/>
        <br/>
        <Button block>Button Block</Button>
        {' '}
        <Button block type="primary">Button Block</Button>
        {' '}
        <Button block type="danger">Button Block</Button>
    </div>
</Playground>

## Prop Types

The following prop types are available for Buttons.

<PropsTable of={Button} />

