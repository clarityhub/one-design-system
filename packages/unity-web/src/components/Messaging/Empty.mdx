---
name: Empty
menu: Messaging
---

import { Component } from 'react';
import { Playground, PropsTable } from 'docz';
import { mdiNotebookOutline } from '@mdi/js';
import Empty from './Empty';
import Button from '../Buttons';

# Empty

Show an Empty message when there are no items to show. The also allows you to
display message, details, and actions.

## \<Empty /\> Component

```js
import { Empty } from '@clarityhub/unity-web/lib/components/Messaging';
```

## Prop Types

### Empty Props

<PropsTable of={Empty} />

## Examples

<Playground>
    <Empty />
</Playground>

<Playground>
    <Empty
        icon={mdiNotebookOutline}
        message="Create your first Notebook"
        details="Notebooks let you organize your customer interviews, notes, and photos together in one place."
        actions={[
            <div>
                <Button type="primary">
                    Create a Notebook
                </Button>
            </div>,
        ]}
    />
</Playground>
