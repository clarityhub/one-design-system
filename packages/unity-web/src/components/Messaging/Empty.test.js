import React from 'react';
import renderer from 'react-test-renderer';
import { mdiNotebookOutline } from '@mdi/js';

import Button from '../Button';
import Empty from './Empty';

describe('<Empty />', () => {
	it('renders default', () => {
		const node = (
			<Empty />
		);
		const tree = renderer.create(node).toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('renders complex', () => {
		const node = (
			<Empty
				icon={mdiNotebookOutline}
				message="Create your first Notebook"
				details="Notebooks let you organize your customer interviews, notes, and photos together in one place."
				actions={[
					<div>
						<Button type="primary">
                            Create a Notebook
						</Button>
					</div>,
				]}
			/>
		);
		const tree = renderer.create(node).toJSON();

		expect(tree).toMatchSnapshot();
	});
});