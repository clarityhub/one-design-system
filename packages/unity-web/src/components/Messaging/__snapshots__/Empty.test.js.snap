// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Empty /> renders complex 1`] = `
.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 2rem 2rem 2rem 2rem;
}

.emotion-2 {
  font-family: Futura,'Trebuchet MS',Arial,sans-serif;
  font-weight: 400;
  font-size: 1.3125rem;
  line-height: 1.2;
  padding-bottom: 1rem;
  padding-top: 3rem;
  color: #322439;
  padding: 0;
  margin-bottom: 0;
  margin-top: 0;
}

.emotion-2 + .emotion-2 {
  margin-bottom: 0;
  margin-top: 0;
}

.emotion-3 {
  font-family: 'Open Sans',Arial,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  color: #322439;
}

.emotion-3 + .emotion-3 {
  margin-top: 1.6rem;
}

.emotion-3 strong,
.emotion-3 b {
  font-weight: 800;
}

.emotion-3 i,
.emotion-3 em {
  font-style: italic;
}

.emotion-4 {
  overflow: visible;
  display: inline-block;
  font-family: 'Open Sans',Arial,sans-serif;
  font-weight: 600;
  -webkit-letter-spacing: 0.35px;
  -moz-letter-spacing: 0.35px;
  -ms-letter-spacing: 0.35px;
  letter-spacing: 0.35px;
  box-shadow: 0 4px 6px rgba(50,50,93,.11),0 1px 3px rgba(0,0,0,.08);
  cursor: pointer;
  text-align: center;
  white-space: nowrap;
  vertical-align: middle;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border: 1px solid transparent;
  padding: .5rem 1.25rem;
  font-size: 1rem;
  line-height: 1.5;
  border-radius: 0.25rem;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  -webkit-transition: color .15s ease-in-out,background-color .15s ease-in-out,border-color .15s ease-in-out,box-shadow .15s ease-in-out;
  transition: color .15s ease-in-out,background-color .15s ease-in-out,border-color .15s ease-in-out,box-shadow .15s ease-in-out;
  --text: #566ad2;
  --text-active: #333;
  --bg: transparent;
  --bg-active: transparent;
  --bg-disabled: transparent;
}

.emotion-4:focus,
.emotion-4:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-4:focus,
.emotion-4.focused {
  outline: 0;
  box-shadow: 0 0 0 .2rem rgba(0,123,255,.25);
}

.emotion-4:disabled,
.emotion-4.disabled {
  cursor: not-allowed;
  pointer-events: none;
}

.emotion-4:not(:disabled):not(.disabled) {
  cursor: pointer;
}

.emotion-4 svg {
  height: 1rem;
  width: 1rem;
}

.emotion-4.-default,
.emotion-4.-primary,
.emotion-4.-danger,
.emotion-4.-white {
  color: var(--text);
  background-color: var(--bg);
  border-color: var(--bg);
}

.emotion-4.-default.-outline,
.emotion-4.-primary.-outline,
.emotion-4.-danger.-outline,
.emotion-4.-white.-outline {
  border-color: var(--outline,var(--bg));
  color: var(--outline-text,var(--bg));
}

.emotion-4.-default.-text,
.emotion-4.-primary.-text,
.emotion-4.-danger.-text,
.emotion-4.-white.-text {
  color: var(--text-text,var(--bg));
}

.emotion-4.-default:hover,
.emotion-4.-primary:hover,
.emotion-4.-danger:hover,
.emotion-4.-white:hover,
.emotion-4.-default.active,
.emotion-4.-primary.active,
.emotion-4.-danger.active,
.emotion-4.-white.active {
  background-color: var(--bg-active);
  border-color: var(--bg-active);
  color: var(--text-active);
}

.emotion-4.-default:hover.-outline,
.emotion-4.-primary:hover.-outline,
.emotion-4.-danger:hover.-outline,
.emotion-4.-white:hover.-outline,
.emotion-4.-default.active.-outline,
.emotion-4.-primary.active.-outline,
.emotion-4.-danger.active.-outline,
.emotion-4.-white.active.-outline {
  background-color: var(--bg) !important;
  border: 1px solid var(--bg);
  color: var(--outline-text-active,var(--text));
}

.emotion-4.-default:hover.-text,
.emotion-4.-primary:hover.-text,
.emotion-4.-danger:hover.-text,
.emotion-4.-white:hover.-text,
.emotion-4.-default.active.-text,
.emotion-4.-primary.active.-text,
.emotion-4.-danger.active.-text,
.emotion-4.-white.active.-text {
  color: var(--text-text-active,var(--bg-active));
}

.emotion-4.-default:disabled,
.emotion-4.-primary:disabled,
.emotion-4.-danger:disabled,
.emotion-4.-white:disabled,
.emotion-4.-default.disabled,
.emotion-4.-primary.disabled,
.emotion-4.-danger.disabled,
.emotion-4.-white.disabled {
  background-color: var(--bg-disabled);
  border-color: var(--bg-disabled);
  color: var(--text);
}

.emotion-4.-default:disabled.-outline,
.emotion-4.-primary:disabled.-outline,
.emotion-4.-danger:disabled.-outline,
.emotion-4.-white:disabled.-outline,
.emotion-4.-default.disabled.-outline,
.emotion-4.-primary.disabled.-outline,
.emotion-4.-danger.disabled.-outline,
.emotion-4.-white.disabled.-outline {
  border: 1px solid var(--outline-disabled,var(--outline-text-disabled,var(--bg-disabled)));
  color: var(--outline-text-disabled,var(--bg-disabled));
}

.emotion-4.-default:disabled.-text,
.emotion-4.-primary:disabled.-text,
.emotion-4.-danger:disabled.-text,
.emotion-4.-white:disabled.-text,
.emotion-4.-default.disabled.-text,
.emotion-4.-primary.disabled.-text,
.emotion-4.-danger.disabled.-text,
.emotion-4.-white.disabled.-text {
  color: var(--text-text-disabled,var(--bg-disabled));
}

.emotion-4.-primary {
  --text: #FFF;
  --text-active: #FFF;
  --outline-disabled: #C2C2C2;
  --bg: #566ad2;
  --bg-active: rgb(70,90,197);
  --bg-disabled: rgb(114,131,211);
}

.emotion-4.-danger {
  --text: #FFF;
  --text-active: #FFF;
  --outline-disabled: #C2C2C2;
  --bg: #f5365c;
  --bg-active: rgb(233,37,76);
  --bg-disabled: rgb(249,139,161);
}

.emotion-4.-default {
  --text: #333;
  --text-active: #333;
  --outline: #333;
  --outline-text: #333;
  --outline-text-active: #333;
  --outline-text-disabled: #C2C2C2;
  --text-text: #333;
  --text-text-active: #333;
  --text-text-disabled: #C2C2C2;
  --bg: #FFF;
  --bg-active: #FDFDFD;
  --bg-disabled: #F1F1F1;
}

.emotion-4.-white {
  --text: #566ad2;
  --text-active: #333;
  --outline: #FFF;
  --outline-text: #FFF;
  --outline-text-active: #566ad2;
  --outline-text-disabled: #C2C2C2;
  --text-text: #FFF;
  --text-text-active: #FFF;
  --text-text-disabled: #C2C2C2;
  --bg: #FFF;
  --bg-active: #FDFDFD;
  --bg-disabled: #F1F1F1;
}

.emotion-4.-text {
  background-color: transparent !important;
  border: 1px solid transparent;
  box-shadow: none;
}

.emotion-4.-text:hover,
.emotion-4.-text.active {
  border-color: transparent;
}

.emotion-4.-text.disabled,
.emotion-4.-text:disabled {
  border: 1px solid transparent;
}

.emotion-4.-outline {
  background-color: transparent !important;
  border: 1px solid inherit;
  box-shadow: none;
}

.emotion-4.disabled,
.emotion-4:disabled {
  cursor: not-allowed;
}

<div
  className="emotion-6 emotion-1"
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <div
    className="emotion-0 emotion-1"
  >
    <svg
      aria-labelledby="icon_labelledby_2"
      style={
        Object {
          "height": "4.5rem",
          "margin": "auto",
          "width": "4.5rem",
        }
      }
      viewBox="0 0 24 24"
    >
      <title
        id="icon_labelledby_2"
      >
        Not found
      </title>
      <path
        d="M17,4V10L15,8L13,10V4H9V20H19V4H17M3,7V5H5V4C5,2.89 5.9,2 7,2H19C20.05,2 21,2.95 21,4V20C21,21.05 20.05,22 19,22H7C5.95,22 5,21.05 5,20V19H3V17H5V13H3V11H5V7H3M5,5V7H7V5H5M5,19H7V17H5V19M5,13H7V11H5V13Z"
        style={
          Object {
            "fill": "currentColor",
          }
        }
      />
    </svg>
  </div>
  <h3
    className="emotion-2"
  >
    Create your first Notebook
  </h3>
  <p
    className="emotion-3"
  >
    Notebooks let you organize your customer interviews, notes, and photos together in one place.
  </p>
  <div>
    <button
      className="-primary emotion-4 emotion-5"
      type="button"
    >
      Create a Notebook
    </button>
  </div>
</div>
`;

exports[`<Empty /> renders default 1`] = `
.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 2rem 2rem 2rem 2rem;
}

.emotion-2 {
  font-family: Futura,'Trebuchet MS',Arial,sans-serif;
  font-weight: 400;
  font-size: 1.3125rem;
  line-height: 1.2;
  padding-bottom: 1rem;
  padding-top: 3rem;
  color: #322439;
  padding: 0;
  margin-bottom: 0;
  margin-top: 0;
}

.emotion-2 + .emotion-2 {
  margin-bottom: 0;
  margin-top: 0;
}

<div
  className="emotion-3 emotion-1"
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  <div
    className="emotion-0 emotion-1"
  >
    <svg
      aria-labelledby="icon_labelledby_1"
      style={
        Object {
          "height": "4.5rem",
          "margin": "auto",
          "width": "4.5rem",
        }
      }
      viewBox="0 0 24 24"
    >
      <title
        id="icon_labelledby_1"
      >
        Not found
      </title>
      <path
        d="M14.46,15.88L15.88,14.46L18,16.59L20.12,14.46L21.54,15.88L19.41,18L21.54,20.12L20.12,21.54L18,19.41L15.88,21.54L14.46,20.12L16.59,18L14.46,15.88M5,3H19C20.11,3 21,3.89 21,5V12.8C20.39,12.45 19.72,12.2 19,12.08V5H5V19H12.08C12.2,19.72 12.45,20.39 12.8,21H5C3.89,21 3,20.11 3,19V5C3,3.89 3.89,3 5,3M7,7H17V9H7V7M7,11H17V12.08C16.15,12.22 15.37,12.54 14.68,13H7V11M7,15H12V17H7V15Z"
        style={
          Object {
            "fill": "currentColor",
          }
        }
      />
    </svg>
  </div>
  <h3
    className="emotion-2"
  >
    No data
  </h3>
</div>
`;
